tests/requirements.txt

```diff
@@ -13,6 +13,6 @@ psutil
 pytest
 pytest-cov
 pytest-mock
-pytest-rerunfailures
+flaky
```

Comment:
We recently switched to pytest-rerunfailures because it's better maintained and used in the conda code base. You should be able to achieve the same as the `rerun_filter` parameter with the `condition` parameter: https://github.com/pytest-dev/pytest-rerunfailures#re-run-individual-failures

Reply:
> We recently switched to pytest-rerunfailures because it's better maintained

Yeah, saw that too, but

> You should be able to achieve the same as the `rerun_filter` parameter with the `condition` parameter: https://github.com/pytest-dev/pytest-rerunfailures#re-run-individual-failures

this appears to be just for the "outside" conditions but doesn't get the test's report information: https://github.com/pytest-dev/pytest-rerunfailures/blob/10.3/pytest_rerunfailures.py#L323-L324
I.e., `pytest-rerunfailures` seems to lack comparable/flexible functionality w.r.t. `flaky`.
AIUI, if we want to use `pytest-rerunfailures`, we have to either be non-selective at all or add `--only-rerun=ConnectionError` to `addopts` in `setup.cfg` and thus be too selective. Since the latter could become a footgun, the former is more sensible.

Do you want me to switch to `pytest-rerunfailures` and thus remove the selectivity on `ConnectionError`?